internal/controllers/topology/cluster/desired_state.go

```diff
@@ -314,6 +316,27 @@ func computeControlPlaneVersion(s *scope.Scope) (string, error) {
 		return *currentVersion, nil
 	}
 
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		// At this point the control plane and the machine deployments are stable and we are almost ready to pick
+		// up the desiredVersion. Call the BeforeClusterUpgrade hook before picking up the desired version.
+		hookRequest := &runtimehooksv1.BeforeClusterUpgradeRequest{
+			Cluster:               *s.Current.Cluster,
+			FromKubernetesVersion: *currentVersion,
+			ToKubernetesVersion:   desiredVersion,
+		}
+		hookResponse := &runtimehooksv1.BeforeClusterUpgradeResponse{}
+		// TODO: consider adding ctx to the functions above the chain?
+		// Will also be useful to pickup a logger from the context to add some log messages.
```

Comment:
Yes, please. As you said we need it for logging anyway sooner or later + it's only computeControlPlaneVersion which doesn't have the context